src/targets/gpu/fuse_mlir.cpp

```diff
@@ -342,37 +342,209 @@ struct find_mlir_fused_ops
         mpm.get_module().replace_instruction(
             ins, mlir_op{gemm_based_op->get_operator()}, inputs, {mm});
     }
+
+    void apply(module_pass_manager& mpm, const match::matcher_result& r) const
+    {
+        auto ins = r.result;
+        auto* pm = ins->module_inputs().front();
+        // Whitelist pointwise operators.
+        if(std::any_of(pm->begin(), pm->end(), [&](const auto& i) {
+               return not is_pointwise_op_supported_by_mlir(i);
+           }))
+            return;
```

Comment:
This should be moved to the matcher:

```cpp
MIGRAPHX_PRED_MATCHER(mlir_pointwise, instruction_ref ins)
{
    if (ins->name() != "pointwise")
        return false;
    auto* pm = ins->module_inputs().front();
    return std::all_of(pm->begin(), pm->end(), [&](const auto& i) {
        return is_pointwise_op_supported_by_mlir(i);
    });
}
```

And then use `mlir_pointwise()` instead of `match::name("pointwise")`.

Reply:
Im going to push_back and say this out of scope and I dont see the reason for this to be done (here and now!) in this PR as this was the pattern used fuse_mlir pass from a long time. 

In principle I dont disagree with you but I feel we can push it out as a cleanup to change such checks through out in fuse_mlir pass not just for attention.